tests/hil/scripts/pytest-hil/board.py

```diff
@@ -1,46 +1,89 @@
 from abc import ABC, abstractmethod
+from contextlib import asynccontextmanager
 import serial
 from time import time, sleep
 import re
 
+import trio
+from trio_serial import SerialStream
+
 class Board(ABC):
-    def __init__(self, port, baud, wifi_ssid, wifi_psk, fw_image, serial_number):
+    def __init__(self, port, baudrate, wifi_ssid, wifi_psk, fw_image, serial_number):
         if serial_number:
             self.serial_number = serial_number
         self.port = port
+        self.baudrate = baudrate
+        self.fw_image = fw_image
+        self.wifi_ssid = wifi_ssid
+        self.wifi_psk = wifi_psk
+
+        self.received: bytes = b''
+
+        self.nursery: trio.Nursery | None = None
+        self.serial: SerialStream | None = None
+
+    @asynccontextmanager
+    async def started(self):
+        async with trio.open_nursery() as nursery:
+            self.nursery = nursery
```

Comment:
Is the idea that we'll use `self.nursery` in a later change to do concurrent send and receive with the native sim process? Or what is the purpose of the nursery here?

Reply:
I've removed `nursery` in the HIL tests with hardware (non-virtual). It seems it is not needed for serial communications with `trio-serial`.